launch/launch/logging/__init__.py

```diff
@@ -30,123 +30,203 @@
-            raise ValueError('{} is not a directory'.format(log_dir))
-        launch_config.log_dir = log_dir
+class LaunchConfig:
+    """Launch Logging Configuration class."""
+
+    def __init__(self):
+        self.reset()
+
+    def reset(self):
+        self._log_dir = None
+        self.file_handlers = {}
+        self.screen_handler = None
+        self.screen_formatter = None
+        self.file_formatter = None
+        self._log_handler_factory = None
+
+    def set_level(self, new_level):
+        """
+        Set up launch logging verbosity level for all loggers.
+
+        :param new_level: the default log level used for all loggers.
+        """
+        logging.root.setLevel(new_level)
+
+    level = property(None, set_level)
+
+    def get_log_dir(self):
+        """Get the current log directory, generating it if necessary."""
+        if self._log_dir is None:
+            self._log_dir = _make_unique_log_dir(
+                base_path=os.path.join(os.path.expanduser('~'), '.ros/log')
+            )
+
+        return self._log_dir
+
+    def set_log_dir(self, new_log_dir):
+        """
+        Set up launch logging directory.
+
+        :param new_log_dir: used as base path for all log file collections.
+        """
+        if new_log_dir is not None:
+            if any(self.file_handlers):
+                import warnings
+                warnings.warn((
+                    'Loggers have been already configured to output to log files below {}. '
+                    'Proceed at your own risk.'
+                ).format(self._log_dir))
+            if not os.path.isdir(new_log_dir):
+                raise ValueError('{} is not a directory'.format(new_log_dir))
+        self._log_dir = new_log_dir
+
+    log_dir = property(get_log_dir, set_log_dir)
+
+    def set_log_handler_factory(self, new_log_handler_factory):
+        """
+        Set up log handler factory.
+
+        :param new_log_handler_factory: a callable to build log file handler.
+           It takes a path to the log file and it must return a `logging.Handler`
+           variant with per logger formatting support.
+           See `launch.logging.handlers` module for further reference and easy reuse
+           of existing standard `logging.handlers` module handlers.
+           Defaults to regular log file handlers for logging if no factory is given.
+        """
+        self._log_handler_factory = new_log_handler_factory
+
+    def get_log_handler_factory(self):
+        """Get the log_handler_factory, generating it if necessary."""
+        if self._log_handler_factory is None:
+            if os.name != 'nt':
+                self._log_handler_factory = handlers.WatchedFileHandler
+            else:
+                self._log_handler_factory = handlers.FileHandler
+        return self._log_handler_factory
+
+    log_handler_factory = property(get_log_handler_factory, set_log_handler_factory)
+
+    def set_screen_format(self, screen_format, screen_style):
```

Comment:
I'll point out that both for `set_screen_format` and `set_log_format`, I got rid of the keyword-only args in favor of traditional args.  I also got rid of default values (essentially `None`) for the arguments.  I slightly prefer this style for small numbers of arguments like this, but I'm willing to change either (or both) decisions.  Let me know what you think.

Reply:
>  I got rid of the keyword-only args in favor of traditional args

I _personally_ think that makes sense for the first, necessary argument but not for the rest e.g. `set_screen_format(self, screen_format, *, screen_style='}')` would be better IMO.